apollo-integration/src/test/java/com/apollographql/apollo/ResponseWriteTestCase.java

```diff
@@ -249,35 +250,63 @@ public void operation_with_fragments() throws Exception {
     assertThat(hero.fragments().heroWithFriendsFragment().friends().get(2).fragments().humanWithIdFragment().id()).isEqualTo("1003");
     assertThat(hero.fragments().heroWithFriendsFragment().friends().get(2).fragments().humanWithIdFragment().name()).isEqualTo("Leia Organa");
 
+    final HumanWithIdFragment supermanAsHumanFragment = new HumanWithIdFragment(
```

Comment:
@sav007 This is an example of https://github.com/apollographql/apollo-android/issues/559

The fact that fragment types are merged in now (and a different type) makes the imperative store api a bit more clumsy and the potential problem -- what happens if a user inconsistently creates the two different fragment objects? Which one will win in the merge process?